knack/config.py

```diff
@@ -47,36 +47,126 @@ def __init__(self, config_dir=None, config_env_var_prefix=None, config_file_name
+        self.config_file_chain.append(_ConfigFile(self.config_dir, self.config_path))
 
     def env_var_name(self, section, option):
         return self._env_var_format.format(section=section.upper(),
                                            option=option.upper())
 
-    def has_option(self, section, option):
+    def has_option(self, section, option, use_local_config=False):
         if self.env_var_name(section, option) in os.environ:
             return True
-        return self.config_parser.has_option(section, option)
+        config_files = self.config_file_chain if use_local_config else self.config_file_chain[-1:]
+        return bool(next((f for f in config_files if f.has_option(section, option)), False))
+
+    def get(self, section, option, fallback=_UNSET, use_local_config=False):
+        env = self.env_var_name(section, option)
+        if env in os.environ:
+            return os.environ[env]
+        last_ex = None
+        for config in self.config_file_chain if use_local_config else self.config_file_chain[-1:]:
+            try:
+                return config.get(section, option)
+            except (configparser.NoSectionError, configparser.NoOptionError) as ex:
+                last_ex = ex
+
+        if fallback is _UNSET:
+            raise last_ex  # pylint:disable=raising-bad-type
+        return fallback
+
+    def items(self, section, use_local_config=False):
+        import re
+        pattern = self.env_var_name(section, '.+')
+        candidates = [(k.split('_')[-1], os.environ[k], k) for k in os.environ.keys() if re.match(pattern, k)]
+        result = {c[0]: c for c in candidates}
+        for config in self.config_file_chain if use_local_config else self.config_file_chain[-1:]:
+            try:
+                entries = config.items(section)
+                for name, value in entries:
+                    if name not in result:
+                        result[name] = (name, value, config.config_path)
+            except (configparser.NoSectionError, configparser.NoOptionError):
+                pass
+        return [{'name': name, 'value': value, 'source': source} for name, value, source in result.values()]
+
+    def getint(self, section, option, fallback=_UNSET, use_local_config=False):
+        return int(self.get(section, option, fallback, use_local_config))
+
+    def getfloat(self, section, option, fallback=_UNSET, use_local_config=False):
+        return float(self.get(section, option, fallback, use_local_config))
+
+    def getboolean(self, section, option, fallback=_UNSET, use_local_config=False):
+        val = str(self.get(section, option, fallback, use_local_config))
+        if val.lower() not in CLIConfig._BOOLEAN_STATES:
+            raise ValueError('Not a boolean: {}'.format(val))
+        return CLIConfig._BOOLEAN_STATES[val.lower()]
 
-    def get(self, section, option, fallback=_UNSET):
+    def set_value(self, section, option, value, use_local_config=False):
```

Comment:
looks  like the `set` method is removed, is it on purpose?

Reply:
Yes, only set_value is needed.